### ISSUE TITLE
Cannot resolve ruby-debug dependencies properly

### ISSUE THREAD

**Issue by rtyler** [jruby-gradle/jruby-gradle-plugin#213]
The [ruby-debug](https://rubygems.org/gems/ruby-debug) gem expresses a version dependency on `ruby-debug-base ~> 0.10.4.0` which JRuby/Gradle complains about:

```
* What went wrong:
Execution failed for task ':run'.
> Could not resolve all dependencies for configuration ':jrubyExec'.
   > Could not find any version that matches rubygems:ruby-debug-base:[0.10.4.0,0.10.4.99999].
     Versions that do not match:
         0.10.6
         0.10.4
         0.10.3
         0.10.2
         0.10.1
         + 7 more
     Required by:
         :test-ruby-debug:unspecified > rubygems:ruby-debug:0.10.4
```

**Comment by mkristian** [comment]
having ```rubygems:ruby-debug-base:[0.10.4.0, 0.10.4.99999]``` as dependency can not resolved to ```rubygems:ruby-debug-base:0.10.4``` via the ivy resolver used by gradle. nothing the plugin can adjust in a sane way, i.e. without increasing the range to [0.10.4,0.10.5] but not work with [0.10.4.2, 0.10.4.99999] ranges anymore.

**Comment by mkristian** [comment]
after a second thought something like:

if more then three version parts then transform the last version parts to one, when last one is '0'

examples:
[0.10.4.0, 0.10.4.99999] => [0.10.4, 0.10.4.99999]
[0.10.5.rc9.0, 0.10.5.rc9.99999] => [0.10.5.rc9, 0.10.5.rc9.99999] 

so basically we can eliminate the trailing 0 digits of a version. this brings us closer to the maven version resolution.
